client.go

```diff
@@ -66,3 +66,98 @@ func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState,
 	}
 	return &updatedFeatureState, nil
 }
+
+func (c *Client) GetProject(projectUUID string) (*Project, error) {
+	url := fmt.Sprintf("%s/projects/", c.baseURL)
+	result := []*Project{}
+	resp, err := c.client.R().
+		SetQueryParams(map[string]string{
+			"uuid": projectUUID,
+		}).
+		SetResult(&result).
+		Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+	if !resp.IsSuccess() || len(result) != 1 {
+		return nil, fmt.Errorf("flagsmithapi: Error getting project: %s", resp)
+	}
+	project := result[0]
+	return project, nil
+
+}
+
+func (c *Client) GetFeature(featureUUID string) (*Feature, error) {
+	url := fmt.Sprintf("%s/features/get-by-uuid/%s/", c.baseURL, featureUUID)
+	feature := Feature{}
+	resp, err := c.client.R().
+		SetQueryParams(map[string]string{
+			"uuid": featureUUID,
+		}).
+		SetResult(&feature).Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !resp.IsSuccess() {
+		return nil, fmt.Errorf("flagsmithapi: Error getting feature: %s", resp)
+	}
+	return &feature, nil
+}
+
+func (c *Client) CreateFeature(feature *Feature) error {
+	projectID := feature.ProjectID
+	if projectID == nil {
+		project, err := c.GetProject(feature.ProjectUUID)
+		if err != nil {
+			return err
+		}
+		projectID = &project.ID
+	}
+
+	url := fmt.Sprintf("%s/projects/%d/features/", c.baseURL, *projectID)
+
+	resp, err := c.client.R().SetBody(feature).SetResult(&feature).Post(url)
+
+	if err != nil {
+		return err
+	}
+
+	if !resp.IsSuccess() {
+		return fmt.Errorf("flagsmithapi: Error creating feature: %s", resp)
+	}
+
+	return nil
+}
+
+func (c *Client) DeleteFeature(projectID, featureID int64) error {
+	url := fmt.Sprintf("%s/projects/%d/features/%d/", c.baseURL, projectID, featureID)
+
+	resp, err := c.client.R().Delete(url)
+
+	if err != nil {
+		return err
+	}
+
+	if !resp.IsSuccess() {
+		return fmt.Errorf("flagsmithapi: Error deleting feature: %s", resp)
+	}
+	return nil
+}
+
+func (c *Client) UpdateFeature(feature *Feature) error {
+	url := fmt.Sprintf("%s/projects/%d/features/%d/", c.baseURL, *feature.ProjectID, *feature.ID)
```

Comment:
If we're checking above to see if projectID is nil, shouldn't we do the same here?

Reply:
The hidden assumption here is that if the user is updating the feature, they must have already read it, hence they will have the Project.ID. but now I think it's better to check and fetch the UUID if it's not present

Reply:
Did you decide against doing this in the end?

Reply:
Ah yes, I lost it during stash/apply